cmake throws error on Ubuntu build

### LVGL version

v9.3.0

### Platform

Ubuntu 22.04

### What happened?

Initially, the cmake failed to find "pcpp" (which was not required previously).  I installed it using "apt install pcc" and now cmake fails with

```
-- The C compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is GNU 11.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python: /usr/bin/python3.10 (found version "3.10.12") found components: Interpreter 
error: unknown option '--passthru-defines'
Error during preprocessing: Command '['pcpp', '-o', '/home/paul/lvgl/build/tmp.h', '--passthru-defines', '--line-directive=', '/home/paul/lvgl/src/lv_conf_internal.h', '-I/home/paul/lvgl', '-I/home/paul/lvgl/..', '-I/home/paul/lvgl/src']' returned non-zero exit status 8.
CMake Error at env_support/cmake/custom.cmake:72 (message):
  preprocess_lv_conf_internal.py failed with return code: 1
Call Stack (most recent call first):
  CMakeLists.txt:23 (include)

```

Reverting this commit
```
commit 371c21bbdaa8216f2a0bef0069db7ae1a727d80d
Author: David Truan <david.truan@edgemtech.ch>
Date:   Thu Apr 3 14:42:18 2025 +0200

    arch(cmake): add native Kconfig support to cmake (#7934)
```

lets the current head build without the error



### How to reproduce?

```
mkdir build ; cd build ; cmake ..
```

I had installed pcpp through "apt install pcc" so I removed this and then did the pip install instead.  
```
$ pcpp --version
pcpp 1.30
```
I usually generate the lv_conf.h by copying the lv_conf_template.h and then editing to change the "#if 0" at the top to "#if 1", so I already had the conf file changes.

The cmake step still fails, but I get a slightly different error:

```
-- Found Python: /usr/bin/python3.10 (found version "3.10.12") found components: Interpreter 
../src/lv_conf_internal.h:60 error: Include file '../../lv_conf.h' not found
Error during preprocessing: Command '['pcpp', '-o', '/home/paul/lvgl/build/tmp.h', '--passthru-defines', '--line-directive=', '/home/paul/lvgl/src/lv_conf_internal.h', '-I/home/paul/lvgl', '-I/home/paul/lvgl/..', '-I/home/paul/lvgl/src']' returned non-zero exit status 1.
CMake Error at env_support/cmake/custom.cmake:72 (message):
  preprocess_lv_conf_internal.py failed with return code: 1
Call Stack (most recent call first):
  CMakeLists.txt:23 (include)
```

The lv_conf.h file is at the same location as always, so not sure why it is not being found.  And as before, if I do

```
git revert 371c21bbdaa8216f
```

to revert that one commit, the cmake and build works without errors.

Thanks for the suggestions!

--- a/env_support/cmake/custom.cmake
+++ b/env_support/cmake/custom.cmake
@@ -34,6 +34,8 @@ file(GLOB_RECURSE THORVG_SOURCES ${LVGL_ROOT_DIR}/src/libs/thorvg/*.cpp ${LVGL_R
 add_library(lvgl ${SOURCES})
 add_library(lvgl::lvgl ALIAS lvgl)
 
+set(PCPP_DEFINITIONS_LIST "")
+
 if ( LV_USE_KCONFIG )
   # kconfig.cmake will generate the .config
   # and autoconf.h, which will be used by lv_conf_kconfig.h
@@ -44,19 +46,27 @@ if ( LV_USE_KCONFIG )
   # generate_cmake_variables.py script
   set(KCONFIG_FLAG --kconfig)
   # If using Kconfig, we need to define additional definitions
-  set(PCPP_ADDITIONAL_DEFS "--defs" "LV_CONF_SKIP" "LV_CONF_KCONFIG_EXTERNAL_INCLUDE=\"${LV_CONF_KCONFIG_EXTERNAL_INCLUDE}\"")
-  else()
+  list(APPEND PCPP_DEFINITIONS_LIST "LV_CONF_SKIP" "LV_CONF_KCONFIG_EXTERNAL_INCLUDE=\"${LV_CONF_KCONFIG_EXTERNAL_INCLUDE}\"")
+else()
   if (LV_CONF_PATH)
-    set(PCPP_ADDITIONAL_DEFS "--defs" "LV_CONF_PATH=\"${LV_CONF_PATH}\"")
+    list(APPEND PCPP_DEFINITIONS_LIST "LV_CONF_PATH=\"${LV_CONF_PATH}\"")
   endif()
+
   if (LV_LVGL_H_INCLUDE_SIMPLE)
-    list(APPEND PCPP_ADDITIONAL_DEFS "--defs" "LV_LVGL_H_INCLUDE_SIMPLE")
+    list(APPEND PCPP_DEFINITIONS_LIST "LV_LVGL_H_INCLUDE_SIMPLE")
   endif()
-  
+
   if (LV_CONF_INCLUDE_SIMPLE)
-    list(APPEND PCPP_ADDITIONAL_DEFS "--defs" "LV_CONF_INCLUDE_SIMPLE")
+    list(APPEND PCPP_DEFINITIONS_LIST "LV_CONF_INCLUDE_SIMPLE")
   endif()
 endif()
+
+if(PCPP_DEFINITIONS_LIST)
+  set(PCPP_ADDITIONAL_DEFS "--defs" ${PCPP_DEFINITIONS_LIST})
+else()
+  set(PCPP_ADDITIONAL_DEFS "")
+endif()
+
 
 target_compile_definitions(
   lvgl PUBLIC $<$<BOOL:${LV_LVGL_H_INCLUDE_SIMPLE}>:LV_LVGL_H_INCLUDE_SIMPLE>

--- a/env_support/cmake/custom.cmake
+++ b/env_support/cmake/custom.cmake
@@ -86,7 +86,7 @@ if(NOT "${ret}" STREQUAL "0")
 endif()
 
 # This will set all CONFIG_LV_USE_* variables in cmake
-include(${CMAKE_BINARY_DIR}/lv_conf.cmake)
+include(${CMAKE_CURRENT_BINARY_DIR}/lv_conf.cmake)
 
 # Add definition of LV_CONF_PATH only if needed
 # Do not redefine it if already defined in tests/CMakeLists.txt

--- a/env_support/cmake/custom.cmake
+++ b/env_support/cmake/custom.cmake
@@ -49,6 +49,13 @@ if ( LV_USE_KCONFIG )
   if (LV_CONF_PATH)
     set(PCPP_ADDITIONAL_DEFS "--defs" "LV_CONF_PATH=\"${LV_CONF_PATH}\"")
   endif()
+  if (LV_LVGL_H_INCLUDE_SIMPLE)
+    list(APPEND PCPP_ADDITIONAL_DEFS "--defs" "LV_LVGL_H_INCLUDE_SIMPLE")
+  endif()
+  
+  if (LV_CONF_INCLUDE_SIMPLE)
+    list(APPEND PCPP_ADDITIONAL_DEFS "--defs" "LV_CONF_INCLUDE_SIMPLE")
+  endif()
 endif()
 
 target_compile_definitions(

